### PR TITLE
 [FrameworkBundle] Add support for setting headers with `TemplateContoller`

### LINKED ISSUE #19909: [FrameworkBundle] Add support for setting `headers` with `TemplateContr…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#54678
| PR author(s) | @HypeMC
| Merged in    | 7.2

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 7.2 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by Timherlaud**
fixes #19909

### CODE REVIEW

File: templates.rst [hunk 1]
```diff
@@ -705,6 +705,11 @@ provided by Symfony:
                     site_name: 'ACME'
                     theme: 'dark'
 
+                # add headers in response
+                headers:
+                    Content-Type: 'image/svg+xml'
+                    foo: 'bar'
```
Comment (alamirault) [inline templates.rst]:
```suggestion
                # optionally you can define HTTP headers to add to the response
                headers:
                    Content-Type: 'text/html'
                    foo: 'bar'
```

File: templates.rst [hunk 2]
```diff
@@ -705,6 +705,11 @@ provided by Symfony:
...
+
     .. code-block:: xml
 
         <!-- config/routes.xml -->
```
Comment (alamirault) [inline templates.rst:715]: Can you also update xml and php examples ?

Comment (alamirault) [inline templates.rst:715]: Foreach new feature, we add directive `versionadded` to tell is new. You can add it above `Checking if a Template Exists` title

```
.. versionadded:: 7.2

    The ``headers`` option was introduced in Symfony 7.2.
```

**Review by alamirault — COMMENTED** [review]
Thanks for your contribution. 

I left some comments

(I didn't see your PR before do it one my side :see_no_evil: , you can take inspiration of mine https://github.com/symfony/symfony-docs/compare/7.1...alamirault:symfony-docs:feature/19909-headers-template-controller)

File: templates.rst (hunk 2 above)
Comment (Timherlaud) [reply templates.rst:715]: done :)

File: templates.rst (hunk 1 above)
Comment (Timherlaud) [reply templates.rst]: done

File: templates.rst (hunk 2 above)
Comment (Timherlaud) [reply templates.rst:715]: Good information :+1:

**Comment by Timherlaud** [comment]
@alamirault thanks for your help, very good work, happy to see how to do a good PR

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Thank you Tim.

**Pull request merged**